scio-avro/src/main/scala/com/spotify/scio/avro/syntax/ScioContextSyntax.scala

```diff
@@ -81,17 +79,13 @@ final class ScioContextOps(private val self: ScioContext) extends AnyVal {
     self.read(SpecificRecordIO[T](path))
 
   /**
-   * Get a typed SCollection from an Avro schema.
+   * Get an SCollection of case classes from an Avro file.
    *
-   * Note that `T` must be annotated with
-   * [[com.spotify.scio.avro.types.AvroType AvroType.fromSchema]],
-   * [[com.spotify.scio.avro.types.AvroType AvroType.fromPath]], or
-   * [[com.spotify.scio.avro.types.AvroType AvroType.toSchema]].
+   * Note that this function uses magnolify's [[magnolify.avro.AvroType]] internally to convert Avro
+   * to case classes.
    */
-  def typedAvroFile[T <: HasAvroAnnotation: TypeTag: Coder](
-    path: String
-  ): SCollection[T] =
-    self.read(AvroTyped.AvroIO[T](path))
+  def typedAvroFile[T: Coder: magnolify.avro.AvroType](path: String): SCollection[T] =
```

Comment:
@farzad-sedghi this may impact the "migration wrapper" code. But this will go to 0.12.2 at least